File: app/src/main/java/com/mlsdev/mlsdevstore/data/model/item/Price.java
package com.mlsdev.mlsdevstore.data.model.item;

import android.os.Parcel;
import android.os.Parcelable;

public class Price implements Parcelable {
    private double value;
    private String currency;

    public Price() {
    }

    public double getValue() {
        return value;
    }

    public String getCurrency() {
        return currency;
    }

    public void setValue(double value) {
        this.value = value;
    }

    public void setCurrency(String currency) {
        this.currency = currency;
    }

    //region Parcelable
    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeDouble(this.value);
        dest.writeString(this.currency);
    }

    protected Price(Parcel in) {
        this.value = in.readDouble();
        this.currency = in.readString();
    }

    public static final Creator<Price> CREATOR = new Creator<Price>() {
        @Override
        public Price createFromParcel(Parcel source) {
            return new Price(source);
        }

        @Override
        public Price[] newArray(int size) {
            return new Price[size];
        }
    };
    //endregion

}

File: app/src/main/java/com/mlsdev/mlsdevstore/data/model/item/Price.kt
package com.mlsdev.mlsdevstore.data.model.item

import android.os.Parcelable
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.mlsdev.mlsdevstore.data.local.database.tables.PricesTable
import kotlinx.android.parcel.IgnoredOnParcel
import kotlinx.android.parcel.Parcelize

@Parcelize
@Entity(tableName = PricesTable.NAME)
data class Price(
        @ColumnInfo(name = PricesTable.COLUMN_PRICE)
        var value: Double = 0.0,
        @ColumnInfo(name = PricesTable.COLUMN_CURRENCY)
        var currency: String? = null
) : Parcelable {

    @IgnoredOnParcel
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = PricesTable.COLUMN_ID)
    var id: Long = 0

}
